packages/experimental/Shadow/src/Shadow.tsx

```diff
@@ -22,12 +21,95 @@ export const Shadow = stagedComponent((props: ShadowProps) => {
       }
     }
 
-    const childWithKeyShadow = React.cloneElement(child, mergeProps(child.props, { style: shadowTokenStyleSet.key }));
+    const { style: childStyle = {}, ...restOfChildProps } = child.props;
```

Comment:
the `= {}` creates a new style object, probably invalidating the memoized function below if the child doesn't have a style on it. You should define the empty style as a const outside of the component so that it's constant.

Reply:
You could also leave it `undefined` and change the memoized function to check for undefined before trying to access properties off of it

Reply:
That was it, thank you Ruriko